From 1f0b8d360ddf8ab91ce04d2e1be529cff9217b9d Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 16 Oct 2018 11:20:28 +0800
Subject: [PATCH 4919/5242] MLK-19947-3 ARM64: dts: freescale: imx8qxp: add
 cpu-idle support

commit  177c44c42ce53e78325dfb788a3e7a7423e1afa0 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds cpu-idle support for i.MX8QXP, since different
platforms have different cpu-idle latency value, so move the
cpu-idle node to platform dtsi.

Add GPT as platform broadcast timer, its clock and power are
managed in TF-A.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Acked-by: Leonard Crestez <leonard.crestez@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8-ca35.dtsi   |    4 +++
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi      |   33 ++++++++++++++++++++
 .../boot/dts/freescale/fsl-imx8qm-mek-inmate.dts   |    2 +-
 .../boot/dts/freescale/fsl-imx8qxp-mek-inmate.dts  |   10 ++++++
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi     |    2 ++
 5 files changed, 50 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8-ca35.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8-ca35.dtsi
index 2a2f155..2af53ca 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8-ca35.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8-ca35.dtsi
@@ -28,6 +28,7 @@
 			reg = <0x0 0x0>;
 			enable-method = "psci";
 			next-level-cache = <&A35_L2>;
+			cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
 		};
 
 		A35_1: cpu@1 {
@@ -36,6 +37,7 @@
 			reg = <0x0 0x1>;
 			enable-method = "psci";
 			next-level-cache = <&A35_L2>;
+			cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
 		};
 
 		A35_2: cpu@2 {
@@ -44,6 +46,7 @@
 			reg = <0x0 0x2>;
 			enable-method = "psci";
 			next-level-cache = <&A35_L2>;
+			cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
 		};
 
 		A35_3: cpu@3 {
@@ -52,6 +55,7 @@
 			reg = <0x0 0x3>;
 			enable-method = "psci";
 			next-level-cache = <&A35_L2>;
+			cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
 		};
 
 		A35_L2: l2-cache0 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 16428d9..65ca3b1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -66,6 +66,30 @@
 		can2 = &flexcan3;
 	};
 
+	cpus {
+		idle-states {
+			entry-method = "psci";
+
+			CPU_SLEEP: cpu-sleep {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x10000>;
+				local-timer-stop;
+				entry-latency-us = <500>;
+				exit-latency-us = <500>;
+				min-residency-us = <5000>;
+			};
+
+			CLUSTER_SLEEP: cluster-sleep {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x10033>;
+				local-timer-stop;
+				entry-latency-us = <500>;
+				exit-latency-us = <2300>;
+				min-residency-us = <14000>;
+			};
+		};
+	};
+
 	memory@80000000 {
 		device_type = "memory";
 		reg = <0x00000000 0x80000000 0 0x40000000>;
@@ -3134,6 +3158,15 @@
 		status = "disabled";
 	};
 
+	gpt0: gpt0@5d140000 {
+		compatible = "fsl,imx8qxp-gpt";
+		reg = <0x0 0x5d140000 0x0 0x4000>;
+		interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&clk IMX8QXP_CLK_DUMMY>, <&clk IMX8QXP_GPT_3M>;
+		clock-names = "ipg", "per";
+		power-domains = <&pd_lsio_gpt0>;
+	};
+
 	dsp: dsp@586e8000 {
 		compatible = "fsl,imx8qxp-dsp";
 		reserved-region = <&dsp_reserved>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-inmate.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-inmate.dts
index 87da05e..147c725 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-inmate.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-inmate.dts
@@ -14,7 +14,7 @@
 
 /dts-v1/;
 #include <dt-bindings/interrupt-controller/arm-gic.h>
-#include "fsl-imx8-ca35.dtsi"
+#include "fsl-imx8-ca53.dtsi"
 #include <dt-bindings/soc/imx_rsrc.h>
 #include <dt-bindings/soc/imx8_hsio.h>
 #include <dt-bindings/soc/imx8_pd.h>
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-inmate.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-inmate.dts
index 09f4992..642f569 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-inmate.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-inmate.dts
@@ -45,6 +45,7 @@
 			compatible = "arm,armv8";
 			enable-method = "psci";
 			reg = <0x0 0x2>;
+			/delete-property/ cpu-idle-states;
 		};
 
 		cpu@1 {
@@ -52,6 +53,15 @@
 			compatible = "arm,armv8";
 			enable-method = "psci";
 			reg = <0x0 0x3>;
+			/delete-property/ cpu-idle-states;
+		};
+
+		cpu@2 {
+			/delete-property/ cpu-idle-states;
+		};
+
+		cpu@3 {
+			/delete-property/ cpu-idle-states;
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index b3c7ed3..d8db692 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -26,6 +26,7 @@
 			reg = <0x0 0x2>;
 			enable-method = "psci";
 			next-level-cache = <&A35_L2>;
+			cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
 		};
 
 		A35_3: cpu@3 {
@@ -34,6 +35,7 @@
 			reg = <0x0 0x3>;
 			enable-method = "psci";
 			next-level-cache = <&A35_L2>;
+			cpu-idle-states = <&CPU_SLEEP &CLUSTER_SLEEP>;
 		};
 	};
 
-- 
1.7.9.5

